### PR TITLE
fix(core/data-tests): Remove duplicates

### LINKED ISSUE #1834: Warn when test appears more than once
Given, for instance:

```HTML
<p data-tests="
  payment-request-show-method.https.html
  some.other.test.html
  payment-request-show-method.https.html">
          ...
</p>
```

ReSpec should warn about duplicate tests. 

Effectively, somewhere here, we need to check and warn of duplicates:
https://github.com/w3c/respec/blob/develop/src/core/data-tests.js#L86-L98

And also drop the duplicates.

### PR DESCRIPTION
**Pull request by Johanna-hub**
Fixes #1834

### CODE REVIEW

File: tests/spec/core/data-tests-spec.js [hunk 1]
```diff
@@ -23,6 +23,13 @@ describe("Core — data-tests attribute", () => {
           >
              metadata tests
           </p>
+        </section>
+        <section>
+          <p id="duplicates" data-tests="
+            payment-request-show-method.https.html,
+            some-other-test.html,
+            payment-request-show-method.https.html">
```
[the remark this answers is not in the document]
Comment (Johanna-hub) [reply tests/spec/core/data-tests-spec.js]: I've added some more repeats in and variation on paths and absolute URLs, just realised I need to follow up with some extra non-repeats.

File: src/core/data-tests.js [hunk 2]
```diff
@@ -88,6 +88,7 @@ export function run(conf) {
       const renderer = hyperHTML.bind(details);
       const testURLs = elem.dataset.tests
         .split(/,/gm)
+        .filter((links, i, self) => self.indexOf(links) === i)
```
[the remark this answers is not in the document]
Comment (Johanna-hub) [reply src/core/data-tests.js]: I have put in the warning, but unsure exactly of the appropriate wording.

File: src/core/data-tests.js [hunk 3]
```diff
@@ -98,12 +98,22 @@ export function run(conf) {
           }
           return href;
         });
+      const duplicates = testURLs.filter(
+        (links, i, self) => self.indexOf(links) !== i
+      );
+      if (duplicates) {
+        pub(
```
Comment (marcoscaceres) [inline src/core/data-tests.js]: We really recently added a new `showInlineWarning` util function that might be good to use here instead. It's shows "inline warnings" on the element itself, which is kinda nice. 

For the message, maybe:
 
```JS
showInlineWarning(
  elem, 
  `Duplicate tests found.`
  `To fix, remove duplicates from "data-tests": ${duplicates.join(", ")}`
);
```

That should add a red squiggly underneath the offending element. See if the message makes sense when you hover over the element.

**Comment by Johanna-hub** [comment]
@marcoscaceres I have made the changes, just one question.  When we display the inline warning, it is showing the generated URLs, this could be different to the original input and what they are looking for to remove.  i.e. they gave `some-other-test.html` and the URL is `https://w3c-test.org/some-API/some-other-test.html` Just checking we can assume they can work it out themselves?

File: tests/spec/core/data-tests-spec.js [hunk 4]
```diff
@@ -138,4 +149,18 @@ describe("Core — data-tests attribute", () => {
...
   });
+  describe("duplicate test removal", () => {
+    it("checks for and removes duplicate links", () => {
+      const items = Array.from(
+        doc.querySelectorAll(
+          "#duplicates > .respec-tests-details > ul > li > a"
+        )
+      );
+      expect(items.length).toEqual(2);
+      expect(items[0].textContent.trim()).toEqual(
```
Comment (marcoscaceres) [inline tests/spec/core/data-tests-spec.js]:
```suggestion
      expect(items[0].textContent.trim()).toBe(
```

File: tests/spec/core/data-tests-spec.js [hunk 5]
```diff
@@ -138,4 +149,18 @@ describe("Core — data-tests attribute", () => {
...
+        "payment-request-show-method"
+      );
+      expect(items[1].textContent.trim()).toEqual("some-other-test");
```
Comment (marcoscaceres) [inline tests/spec/core/data-tests-spec.js]:
```suggestion
      expect(items[1].textContent.trim()).toBe("some-other-test");
```

**Pull request merged**